### PR TITLE
Reworks skittish quirk to be automatic

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
tweak: The Skittish quirk will now cause you to automatically dive into
a locker/crate if you bump into it (although closets must be closed).
Walk to avoid this behaviour.
/:cl:

This makes the quirk more useful, while also making it more thematic,
since the "diving into" behaviour can't be disabled, only supressed by
walking.

The cost is unchanged, as the quirk in its current form is overcosted at
2 points.

The emergent effect of skittish people diving into closets when caught
into explosions is definitely a feature, and not a bug.

---

~~There is some performance work that can be done, (moving the bumping behaviour~~
~~from the closet to the mob), but only if this is popular.~~

The "skittish" code is now an element that is added/removed from /mob/livings when they
gain/lose the TRAIT_SKITTISH trait.

It handles means that instead of listening for all atoms that bump into closets, we instead
only check when a skittish person bumps into a thing.

### CODE REVIEW

**Comment by cacogen** [comment]
Not sure how practical it would be but it's interesting

**Comment by ATH1909** [comment]
"tweak: The Skittish quirk will now cause you to automatically dive into
a locker/crate if you move into it while it is closed. Walk to avoid
this behaviour."
What if the locker/crate is open?

**Comment by coiax** [comment]
> 
> 
> "tweak: The Skittish quirk will now cause you to automatically dive into
> a locker/crate if you move into it while it is closed. Walk to avoid
> this behaviour."
> What if the locker/crate is open?

Good point. It works if you "Bump" into the crate/locker. So it works if you run into a crate, regardless of it being open or closed, but only worked on closed lockers. I will update the CL.

File: code/game/objects/structures/crates_lockers/closets.dm [hunk 1]
```diff
@@ -105,7 +105,7 @@
 	if(isliving(user))
 		var/mob/living/L = user
 		if(HAS_TRAIT(L, TRAIT_SKITTISH))
-			. += "<span class='notice'>Ctrl-Shift-click [src] to jump inside.</span>"
+			. += "<span class='notice'>If you bump into [src] while running, you will jump inside.</span>"
```
Comment (Fikou) [inline code/game/objects/structures/crates_lockers/closets.dm:108]: i think simply "it" instead of [src] would sound better here?

Comment (Jared-Fogle) [reply code/game/objects/structures/crates_lockers/closets.dm:108]: `[src.p_them()]`

**Comment by Pugie** [comment]
Why, it worked fine before?
At least until rohesie broke it with knockdown timer.

File: code/game/objects/structures/crates_lockers/closets.dm [hunk 2]
```diff
@@ -37,6 +37,7 @@
 	var/delivery_icon = "deliverycloset" //which icon to use when packagewrapped. null to be unwrappable.
 	var/anchorable = TRUE
 	var/icon_welded = "welded"
+	var/divable = TRUE /// whether a skittish person can dive inside this closet. disable if opening the closet causes "bad things" to happen.
```
Comment (Rohesie) [inline code/game/objects/structures/crates_lockers/closets.dm:40]:
```suggestion
	/// Whether a skittish person can dive inside this closet. disable if opening the closet causes "bad things" to happen.
	var/divable = TRUE
```

**Comment by Rohesie** [comment]
See if you can address the linter's issue.

File: code/game/objects/structures/crates_lockers/closets.dm (hunk 2 above)
Comment (cacogen) [reply code/game/objects/structures/crates_lockers/closets.dm:40]: Also capitalise disable

File: code/datums/elements/skittish.dm [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+/datum/element/skittish/Attach(datum/target)
+	. = ..()
+	if(!isliving(target))
+		return ELEMENT_INCOMPATIBLE
+
+	RegisterSignal(target, COMSIG_MOVABLE_BUMP, .proc/Bump)
+
+/datum/element/skittish/Detach(datum/target)
+	. = ..()
+	UnregisterSignal(target, COMSIG_MOVABLE_BUMP)
```
Comment (Rohesie) [inline code/datums/elements/skittish.dm:17]:
```suggestion
/datum/element/skittish/Detach(datum/target)
	UnregisterSignal(target, COMSIG_MOVABLE_BUMP)
	return ..()
```

**Comment by coiax** [comment]
Looks like my VM's clock is wrong, so it appears I did these commits yesterday, but I did them just now.

File: code/datums/traits/good.dm [hunk 4]
```diff
@@ -190,6 +190,7 @@
 	name = "Skittish"
 	desc = "You're easy to startle, and hide frequently. Run into a closed locker to jump into it, as long as you have access. You can walk to avoid this."
 	value = 2
+	value = 8
```
Comment (Jared-Fogle) [inline code/datums/traits/good.dm]: two values 😊

**Pull request merged**